open_src/influx/httpd/handler.go

```diff
@@ -533,6 +551,16 @@ func (h *Handler) serveQuery(w http.ResponseWriter, r *http.Request, user meta2.
 		h.Logger.Error("query error! parsing query value", zap.Error(err), zap.String("db", db), zap.Any("r", r))
 		return
 	}
+
+	// if subscription is not enabled,
+	// and the query contains subscription statements,
+	// then return error
+	if h.SubscriberManager == nil && containSubscriptionStatements(q.Statements) {
```

Comment:
Doing this judgment on the query hot path will greatly reduce qps.

Reply:
> Doing this judgment on the query hot path will greatly reduce qps.
Yes, I need to use other method to disable the subscription statements when subscription is not enabled.